### PR TITLE
Gradle Support

### PR DESCRIPTION
**Pull request by dshimo**
`gradle build` to build and test
`gradle clean` to clean everything (build and target dirs)
`gradle libertyStart` to start servers
`gradle libertyStop` to stop servers

### CODE REVIEW

**Comment by dshimo** [comment]
Working on docs, Where's the travis check?

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+subprojects {
+    apply plugin: 'java'
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+    }
+
+    repositories {
+        mavenLocal()
```
Comment (cthigh) [inline build.gradle]: Can't this be removed?

File: microservice-schedule/build.gradle [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    compile project(':demo-bootstrap')
+    providedCompile group: 'javax', name: 'javaee-api', version:'7.0'
+}
+
+buildscript {
+	repositories {
+		mavenLocal()
+		mavenCentral()
+		maven {
+			url 'https://oss.sonatype.org/content/repositories/snapshots/'
```
Comment (cthigh) [inline microservice-schedule/build.gradle]: use the 2.0.1-SNAPSHOT

File: microservice-session/build.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+description = 'Conference :: Session'
+dependencies {
+    compile project(':demo-bootstrap')
+    providedCompile group: 'javax', name: 'javaee-api', version:'7.0'
+    providedCompile group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version:'2.8.2'
+}
+
+buildscript {
+	repositories {
```
Comment (cthigh) [inline microservice-session/build.gradle]: correct the tabs.

File: microservice-session/build.gradle [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		mavenLocal()
```
Comment (cthigh) [inline microservice-session/build.gradle]: remove mavenLocal()

File: microservice-session/build.gradle [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		mavenCentral()
+		maven {
+			url 'https://oss.sonatype.org/content/repositories/snapshots/'
+		}
+	}
+	dependencies {
+		classpath 'net.wasdev.wlp.gradle.plugins:liberty-gradle-plugin:2.1-SNAPSHOT'
```
Comment (cthigh) [inline microservice-session/build.gradle]: change to 2.0.1-SNAPSHOT

File: microservice-speaker/build.gradle [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    providedCompile group: 'javax.enterprise', name: 'cdi-api', version:'1.2'
+    providedCompile group: 'javax.ws.rs', name: 'javax.ws.rs-api', version:'2.0'
+
+    // Include tools.jar in classpath
+    compile files("${System.properties['java.home']}/../lib/tools.jar")
+}
+
+buildscript {
+    repositories {
+        mavenLocal()
```
Comment (cthigh) [inline microservice-speaker/build.gradle]: remove

File: microservice-speaker/build.gradle [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        mavenCentral()
+        maven {
+            url 'https://oss.sonatype.org/content/repositories/snapshots/'
+        }
+    }
+    dependencies {
+        classpath 'net.wasdev.wlp.gradle.plugins:liberty-gradle-plugin:2.1-SNAPSHOT'
```
Comment (cthigh) [inline microservice-speaker/build.gradle]: 2.0.1-SNAPSHOT

File: microservice-vote/build.gradle [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    compileOnly group: 'javax.inject', name: 'javax.inject', version:'1'
+    compileOnly group: 'javax.interceptor', name: 'javax.interceptor-api', version: '1.2.1'
+    compileOnly group: 'javax.enterprise', name: 'cdi-api', version: '2.0'
+    providedCompile group: 'javax.annotation', name: 'javax.annotation-api', version:'1.2'
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version:'17.0.0.3'
+}
+
+buildscript {
+	repositories {
+		mavenLocal()
```
Comment (cthigh) [inline microservice-vote/build.gradle]: tabs

File: microservice-vote/build.gradle [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+war {
+	archiveName = 'ROOT.' + extension
+}
+
+ext {
+	// Liberty server properties
+    wlpServerName = 'voteServer'
+	testServerHttpPort = 7070
```
Comment (cthigh) [inline microservice-vote/build.gradle]: spacing

File: microservice-vote/build.gradle [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    testServerHttpsPort = 9443
+    warContext = "vote"
+}
+
+liberty {
+	server {
+		name = wlpServerName
```
Comment (cthigh) [inline microservice-vote/build.gradle]: tabs.

File: microservice-vote/build.gradle [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+		bootstrapProperties = ['default.http.port': testServerHttpPort, 'default.https.port': testServerHttpsPort]
+        dropins = [war]
+        looseApplication = false
+	}
```
Comment (cthigh) [inline microservice-vote/build.gradle]: formatting.

File: microservice-vote/build.gradle [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+    systemProperties = ['liberty.test.port': testServerHttpPort, 'war.context': warContext]
+}
+
+task url {
+    doLast { println 'Vote :: http://localhost:7070/' }
+}
+
+// libertyRun.dependsOn 'installApps'
+// libertyStart.dependsOn 'installApps'
```
Comment (cthigh) [inline microservice-vote/build.gradle]: remove unneeded depends

File: web-application/build.gradle [hunk 13]
```diff
@@ -0,0 +1,100 @@
+apply plugin: 'war'
+apply plugin: 'liberty'
+// apply plugin: 'org.ysb33r.nodejs.base'
+// apply plugin: 'org.ysb33r.nodejs.npm'
```
Comment (cthigh) [inline web-application/build.gradle]: remove

File: microservice-vote/build.gradle [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+    compile project(':demo-bootstrap')
+    testCompile group: 'org.glassfish.jersey.core', name: 'jersey-common', version:'2.23.2'
+    testCompile group: 'org.apache.cxf', name: 'cxf-rt-rs-client', version:'3.1.1'
+    testCompile group: 'org.glassfish', name: 'javax.json', version:'1.0.4'
+    compileOnly group: 'javax.ws.rs', name: 'javax.ws.rs-api', version:'2.0.1'
+    compileOnly group: 'javax.inject', name: 'javax.inject', version:'1'
+    compileOnly group: 'javax.interceptor', name: 'javax.interceptor-api', version: '1.2.1'
+    compileOnly group: 'javax.enterprise', name: 'cdi-api', version: '2.0'
+    providedCompile group: 'javax.annotation', name: 'javax.annotation-api', version:'1.2'
+    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version:'17.0.0.3'
```
Comment (cthigh) [inline microservice-vote/build.gradle]: seems like all subprojects should use open liberty.

File: microservice-schedule/build.gradle [hunk 15]
```diff
@@ -0,0 +1,43 @@
+apply plugin: 'liberty'
+apply plugin: 'war'
+
+description = 'Conference :: Schedule'
+dependencies {
+    compile project(':demo-bootstrap')
+    providedCompile group: 'javax', name: 'javaee-api', version:'7.0'
```
Comment (cthigh) [inline microservice-schedule/build.gradle]: libertyRuntime

File: microservice-session/build.gradle [hunk 16]
```diff
@@ -0,0 +1,45 @@
+apply plugin: 'liberty'
+apply plugin: 'war'
+
+description = 'Conference :: Session'
+dependencies {
+    compile project(':demo-bootstrap')
+    providedCompile group: 'javax', name: 'javaee-api', version:'7.0'
+    providedCompile group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version:'2.8.2'
```
Comment (cthigh) [inline microservice-session/build.gradle]: libertyRuntime

File: microservice-speaker/build.gradle [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+    testCompile group: 'net.javacrumbs.json-unit', name: 'json-unit-fluent', version:'1.12.1'
+    testCompile group: 'org.jboss.resteasy', name: 'resteasy-jackson-provider', version:'3.0.19.Final'
+    testCompile group: 'org.jboss.resteasy', name: 'resteasy-client', version:'3.0.19.Final'
+    testCompile group: 'org.jboss.shrinkwrap', name: 'shrinkwrap-api', version: '1.2.6'
+    testCompile group: 'org.jboss.shrinkwrap.resolver', name: 'shrinkwrap-resolver-impl-maven', version:'2.2.0'
+    providedCompile group: 'javax.enterprise', name: 'cdi-api', version:'1.2'
+    providedCompile group: 'javax.ws.rs', name: 'javax.ws.rs-api', version:'2.0'
+
+    // Include tools.jar in classpath
+    compile files("${System.properties['java.home']}/../lib/tools.jar")
```
Comment (cthigh) [inline microservice-speaker/build.gradle]: libertyRuntime

**Pull request merged**